Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -1193,24 +1203,26 @@ void FitPeaks::fitSpectrumPeaks(
       cost =
           fitIndividualPeak(wi, peak_fitter, expected_peak_pos, peak_window_i,
                             observe_peak_width, peakfunction, bkgdfunction);
-      if (cost < 1e7) { // assume it worked and save out the result
-        // reset the flag such that there is at a peak fit in this spectrum
-        neighborPeakSameSpectrum = true;
-        // copy values
-        for (size_t i = 0; i < lastGoodPeakParameters[peak_index].size(); ++i) {
-          lastGoodPeakParameters[peak_index][i] = peakfunction->getParameter(i);
-          localPrevGoodResults[i] = peakfunction->getParameter(i);
-        }
-      }
     }
 
     // process fitting result
     FitPeaksAlgorithm::FitFunction fit_function;
     fit_function.peakfunction = peakfunction;
     fit_function.bkgdfunction = bkgdfunction;
 
-    processSinglePeakFitResult(wi, peak_index, cost, expected_peak_centers,
-                               fit_function, fit_result); // sets the record
+    auto good_fit =
```

Comment:
Great!  I tried in my last PR to move this section out of the nested if-else but resulted in a much poorer fitting. It seems that you have the issue fixed!